workspaces/arborist/lib/arborist/build-ideal-tree.js

```diff
@@ -737,8 +737,7 @@ This is a one-time fix-up, please be patient...
         const id = useResolved ? resolved
           : version || `file:${node.path}`
         const spec = npa.resolve(name, id, dirname(path))
-        const sloc = location.substr('node_modules/'.length)
-        const t = `idealTree:inflate:${sloc}`
+        const t = `idealTree:inflate:${location}`
```

Comment:
If we really care about stripping leading node_modules we can
```suggestion
        const t = `idealTree:inflate:${location.replace(/^node_modules\//, '')}`
```